client/src/details/UrlBuilder.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+
+// Forward declaration
+struct Curl_URL;
+typedef struct Curl_URL CURLU;
+
+namespace SFS::details
+{
+class ReportingHandler;
+
+enum class Scheme
+{
+    Https,
+};
+
+class UrlBuilder
+{
+  public:
+    UrlBuilder(const ReportingHandler& handler);

```

Comment:
You don't want to have a constructor that takes a string-URL? It could help with the common case of creating UrlBuilder when caller already has a string-URL (no need to call SetUrl immediately after object creation).